core/src/main/scala/kafka/tools/ConsoleProducer.scala

```diff
@@ -284,6 +284,12 @@ object ConsoleProducer {
     val maxPartitionMemoryBytes = options.valueOf(maxPartitionMemoryBytesOpt)
     val metadataExpiryMs = options.valueOf(metadataExpiryMsOpt)
     val maxBlockMs = options.valueOf(maxBlockMsOpt)
+    if (!useOldProducer && options.has(keyEncoderOpt) && keyEncoderClass != "org.apache.kafka.common.serialization.ByteArraySerializer") {
+      throw new KafkaException(s"Unexpected serializer for key-serializer. Supplied $keyEncoderOpt, allowed: org.apache.kafka.common.serialization.ByteArraySerializer")
+    }
+    if (!useOldProducer  && options.has(keyEncoderOpt) && valueEncoderClass != "org.apache.kafka.common.serialization.ByteArraySerializer") {
```

Comment:
@aishraj I think it should be valueEncoderOpt instead of options.has(keyEncoderOpt), since you are validating for valueEncoderClass.